lib/streamlit/state/session_state.py

```diff
@@ -126,3 +126,40 @@ def __delattr__(self, key: str) -> None:
             del self[key]
         except KeyError:
             raise AttributeError(key)
+
+
+def _get_current_session() -> "ReportSession":
+    # Getting the session id easily comes from the report context, which is
+    # a little weird, but a precedent that has been set.
+    ctx = get_report_ctx()
+    this_session: Optional["ReportSession"] = None
+
+    if ctx is not None:
+        this_session = Server.get_current().get_session_by_id(ctx.session_id)
+
+    if this_session is None:
+        raise RuntimeError(
+            "We were unable to retrieve your Streamlit session."
+            " Is your application utilizing threads? It's possible that could"
+            " be conflicting with our system."
+        )
+
+    return this_session
+
+
+def get_session_state() -> SessionState:
```

Comment:
Should we add a leading underscore, to emphasize that it is an internal function?

Reply:
I actually realized we should do this in the upcoming-PR pulling in `LazySessionState` from the prototype, but since I'm making some changes to this PR I'll include it here